File: VideoPlayerAPI/BusinessLogic/Videos/Models/Video.cs
using VideoPlayerAPI.BusinessLogic.Users.Models;

namespace VideoPlayerAPI.BusinessLogic.Videos.Models;

public class Video
{
    public Guid Id { get; init; }
    public string Title { get; init; }
    public string Description { get; init; }
    public string VideoUrl { get; init; }
    public string ThumbnailUrl { get; init; }
    public DateTimeOffset UploadDate { get; init; }
    public TimeSpan Duration { get; init; }
    public string UploadDateInfo { get; init; }
    public User UploadedBy { get; init; }
    public int Views { get; set; }
    public int Likes { get; set; }
    public int Dislikes { get; set; }

    public void IncrementViews()
    {
        Views++;
    }
}

File: VideoPlayerClient/src/app/core/_models/video.ts
import { User } from './user';

export interface Video {
  id: string;
  title: string;
  description: string;
  videoUrl: string;
  thumbnailUrl: string;
  uploadDate: string;
  duration: string;
  uploadDateInfo: string;
  uploadedBy: User;
  views: number;
  likes: number;
  dislikes: number;
}

export interface UploadVideo {
  title: string;
  description: string;
  videoData: VideoData;
  requestedById: string;
}

export interface VideoData {
  videoType: string;
  bytes: string;
}
